### DISCUSSION TITLE
What is the behavior of InteractionManager on web?

### DISCUSSION THREAD

**Discussion by g33kracc00n** [necolas/react-native-web#1943] [Q&A]
Hello, 👨🏻‍💻

do we have a specified or documented behavior of InteractionManager on web ?
and what about runAfterInteractions ? does it really wait for interactions to finish before executing the passed code (on web)?

This note is on React Native documentation:

**Note: InteractionManager.runAfterInteractions() is not working properly on web. It triggers immediately without waiting until the interaction is finished.**

[https://reactnative.dev/docs/interactionmanager#runafterinteractions](url)

**Comment by intergalacticspacehighway** [comment]
@MohamedDarwiche-DaherSoftware On web, it's using [requestIdleCallback](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback). You can [verify it here](https://github.com/necolas/react-native-web/blob/master/packages/react-native-web/src/exports/InteractionManager/index.js#L27).

So, if the browser supports requestIdleCallback, it should run when the thread is idle.
Seems like [Safari still lacks the support](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback#browser_compatibility)

If browser doesn't support, it uses [setTimeout as a fallback](https://github.com/necolas/react-native-web/blob/master/packages/react-native-web/src/modules/requestIdleCallback/index.js) which will also wait for synchronous tasks to finish.
